binding/HarfBuzzSharp.Shared/FontFunctions.cs

```diff
@@ -0,0 +1,553 @@
+			}
+
+			if (del == null) {
+				throw new ArgumentException (nameof (del));
+			}
+
+			var wrappedDelegate = new FontExtentsDelegate ((Font _, object __, out FontExtents e) =>
+				del.Invoke (Font, FontData, out e));
+
+			var ctx = DelegateProxies.CreateMulti (wrappedDelegate, context, destroy);
+
+			HarfBuzzApi.hb_font_funcs_set_font_h_extents_func (Handle, FontExtentsProxy, ctx,
+				DelegateProxies.ReleaseDelegateProxyForMulti);
+		}
+
+		public void SetVerticalFontExtentsDelegate (FontExtentsDelegate del, object context = null,
+			ReleaseDelegate destroy = null)
+		{
+			if (IsImmutable) {
+				throw new NotSupportedException ($"{nameof (FontFunctions)} is immutable and can't be changed.");
+			}
+
+			if (del == null) {
+				throw new ArgumentException (nameof (del));
+			}
+
+			var wrappedDelegate = new FontExtentsDelegate ((Font _, object __, out FontExtents e) =>
+				del.Invoke (Font, FontData, out e));
+
+			var ctx = DelegateProxies.CreateMulti (wrappedDelegate, context, destroy);
+
+			HarfBuzzApi.hb_font_funcs_set_font_v_extents_func (Handle, FontExtentsProxy, ctx,
+				DelegateProxies.ReleaseDelegateProxyForMulti);
+		}
+
+		[MonoPInvokeCallback (typeof (NominalGlyphProxyDelegate))]
+		private static bool NominalGlyphProxyImplementation (IntPtr font, IntPtr fontData, uint unicode, out uint glyph, IntPtr context)
+		{
+			var multi = DelegateProxies.Get<GetMultiDelegateDelegate> (context, out _);
+			var del = (NominalGlyphDelegate)multi.Invoke (typeof (NominalGlyphDelegate));
+			return del.Invoke (null, null, unicode, out glyph);
+		}
+
+		public void SetNominalGlyphDelegate (NominalGlyphDelegate del, object context = null,
+			ReleaseDelegate destroy = null)
+		{
+			if (IsImmutable) {
+				throw new NotSupportedException ($"{nameof (FontFunctions)} is immutable and can't be changed.");
+			}
+
+			if (del == null) {
+				throw new ArgumentException (nameof (del));
+			}
+
+			var wrappedDelegate = new NominalGlyphDelegate ((Font _, object __, uint u, out uint g) =>
+				del.Invoke (Font, FontData, u, out g));
+
+			var ctx = DelegateProxies.CreateMulti (wrappedDelegate, context, destroy);
+
+			HarfBuzzApi.hb_font_funcs_set_nominal_glyph_func (Handle, NominalGlyphProxy, ctx,
+				DelegateProxies.ReleaseDelegateProxyForMulti);
+		}
+
+		[MonoPInvokeCallback (typeof (NominalGlyphsProxyDelegate))]
+		private static uint NominalGlyphsProxyImplementation (IntPtr font, IntPtr fontData, uint count,
+			IntPtr firstUnicode, uint unicodeStride, out IntPtr firstGlyph, out uint glyphStride, IntPtr context)
+		{
+			var multi = DelegateProxies.Get<GetMultiDelegateDelegate> (context, out _);
+			var del = (NominalGlyphsDelegate)multi.Invoke (typeof (NominalGlyphsDelegate));
+
+			unsafe {
+				var unicodes = new ReadOnlySpan<uint> ((void*)firstUnicode, (int)unicodeStride);
+				var glyphCount = del.Invoke (null, null, count, unicodes, out var glyphs);
+
+				fixed (uint* glyphsPtr = glyphs) {
```

Comment:
Not sure how to do this properly. If the delegate creates an array on native side everything is fine but if the array is managed this will break.